packages/e2e-tests/src/SpawnSync.ts

```diff
@@ -0,0 +1,105 @@
+import * as fs from 'fs';
+import * as core from '@actions/core';
+import * as child_process from 'child_process';
+import type {
+	ProcessEnvOptions,
+	StdioOptions,
+	IOType,
+	SpawnSyncReturns,
+	SpawnSyncOptions,
+	SpawnSyncOptionsWithStringEncoding,
+} from 'child_process';
+
+type Call = (
+	command: string,
+	args: string[],
+	opts: {
+		input?: SpawnSyncOptions['input'];
+		stdin?: IOType;
+		stdout?: IOType;
+		env?: ProcessEnvOptions['env'];
+	}
+) => SpawnSyncReturns<string>;
+
+interface ExecSyncInterface {
+	call: Call;
+}
```

Comment:
yikes, did you have any experience with any of this before working with EE?
or did you have to figure out all of the above just for this PR?

Reply:
I had very limited experience working with pipes in Node before this PR 😅 After this PR though, it's more or clear in terms of how to handle them in Node. When to use `pipe` or when to `inherit`. When to use `shell` and when not, etc.